core/debug/src/debug.c

```diff
@@ -230,3 +230,117 @@ void debug_fault(THaltError reason, uint32_t lr, uint32_t sp)
     debug_mpu_config();
     DEBUG_PRINT_END();
 }
+
+static void __debug_reboot(void)
+{
+    UVISOR_SVC(UVISOR_SVC_ID_DEBUG_REBOOT, "");
+}
+
+void debug_reboot(void)
+{
+    if (!g_debug_box.initialized || g_active_box != g_debug_box.box_id) {
+        HALT_ERROR(NOT_ALLOWED, "This function can only be called from the context of an initialized debug box.\n\r");
+    }
+
+    /* Reboot.
+     * This can only be called from privileged code. */
+    NVIC_SystemReset();
```

Comment:
Added comment that explains this better. This anyway *cannot* be called from unprivileged code. Unprivileged code will run `__debug_reboot` instead (without knowing -- It's part of the stack tampering process, see [this line](https://github.com/ARMmbed/uvisor/pull/178/files#diff-8866e0e2dcd8960e585a789e8e817ff1R288)). `__debug_reboot` then calls the proper SVCall.